library/lua/gui.lua

```diff
@@ -955,19 +983,35 @@ INTERIOR_MEDIUM_FRAME = FRAME_INTERIOR_MEDIUM
 function paint_frame(dc,rect,style,title,inactive,pause_forced,resizable)
     local pen = style.frame_pen
     local x1,y1,x2,y2 = dc.x1+rect.x1, dc.y1+rect.y1, dc.x1+rect.x2, dc.y1+rect.y2
-    dscreen.paintTile(style.lt_frame_pen or pen, x1, y1)
-    dscreen.paintTile(style.rt_frame_pen or pen, x2, y1)
-    dscreen.paintTile(style.lb_frame_pen or pen, x1, y2)
-    local rb_frame_pen = style.rb_frame_pen
-    if rb_frame_pen == FRAME_WINDOW.rb_frame_pen and not resizable then
-        rb_frame_pen = FRAME_PANEL.rb_frame_pen
+    local signed = false
+
+    if x1 == x2 and y1 == y2 then
+        dscreen.paintTile(style.x_frame_pen or pen, x1, y2)
+    elseif x1 == x2 then
+        dscreen.paintTile(style.tte_frame_pen or style.tti_frame_pen or pen, x1, y1)
+        dscreen.paintTile(style.bte_frame_pen or style.bti_frame_pen or pen, x2, y2)
+        dscreen.fillRect(style.v_frame_pen or pen, x1,y1+1,x2,y2-1)
+    elseif y1 == y2 then
+        dscreen.paintTile(style.lte_frame_pen or style.lti_frame_pen or pen, x1, y1)
+        dscreen.paintTile(style.rte_frame_pen or style.rti_frame_pen or pen, x2, y2)
+        dscreen.fillRect(style.h_frame_pen or pen, x1+1,y1,x2-1,y2)
+    else
+        dscreen.paintTile(style.lt_frame_pen or pen, x1, y1)
+        dscreen.paintTile(style.rt_frame_pen or pen, x2, y1)
+        dscreen.paintTile(style.lb_frame_pen or pen, x1, y2)
+        local rb_frame_pen = style.rb_frame_pen
+        if rb_frame_pen == FRAME_WINDOW.rb_frame_pen and not resizable then
+            rb_frame_pen = FRAME_PANEL.rb_frame_pen
+        end
+        dscreen.paintTile(rb_frame_pen or pen, x2, y2)
+        dscreen.fillRect(style.t_frame_pen or style.h_frame_pen or pen,x1+1,y1,x2-1,y1)
+        dscreen.fillRect(style.b_frame_pen or style.h_frame_pen or pen,x1+1,y2,x2-1,y2)
+        dscreen.fillRect(style.l_frame_pen or style.v_frame_pen or pen,x1,y1+1,x1,y2-1)
+        dscreen.fillRect(style.r_frame_pen or style.v_frame_pen or pen,x2,y1+1,x2,y2-1)
+        signed = style.signature_pen
     end
-    dscreen.paintTile(rb_frame_pen or pen, x2, y2)
-    dscreen.fillRect(style.t_frame_pen or style.h_frame_pen or pen,x1+1,y1,x2-1,y1)
-    dscreen.fillRect(style.b_frame_pen or style.h_frame_pen or pen,x1+1,y2,x2-1,y2)
-    dscreen.fillRect(style.l_frame_pen or style.v_frame_pen or pen,x1,y1+1,x1,y2-1)
-    dscreen.fillRect(style.r_frame_pen or style.v_frame_pen or pen,x2,y1+1,x2,y2-1)
-    if style.signature_pen ~= false then
+
+    if signed ~= false then
```

Comment:
working on this `signed` thing made me realise that we don't need a whole new FRAME_INTERIOR_MEDIUM for planneroverlay... 🤦 i could've just turned off the signature_pen for _that_ particular panel... so expect PRs for that as well.

but anyway- this code basically only renders the `DFHack` thingy if the panel has a width/height > 1.